applications/teletype/src/templates/teletype_voicemail_deleted.erl

```diff
@@ -0,0 +1,312 @@
+
+    case kz_datamgr:fetch_attachment(Db, {kzd_box_message:type(), VMId}, AttachmentId) of
+        {'ok', Bin} -> [{ContentType, FileName, Bin}];
+        {'error', _E} ->
+            lager:debug("failed to fetch attachment ~s: ~p", [AttachmentId, _E]),
+            throw({'error', 'no_attachment'})
+    end.
+
+-spec get_file_name(kz_json:object(), kz_term:proplist()) -> kz_term:ne_binary().
+get_file_name(MediaJObj, Macros) ->
+    %% CallerID_Date_Time.mp3
+    CallerID =
+        case {props:get_value([<<"caller_id">>, <<"name">>], Macros)
+             ,props:get_value([<<"caller_id">>, <<"number">>], Macros)
+             }
+        of
+            {'undefined', 'undefined'} -> props:get_value([<<"from">>, <<"user">>], Macros, <<"Unknown">>);
+            {'undefined', Num} -> Num;
+            {Name, _} -> Name
+        end,
+
+    LocalDateTime = props:get_value([<<"date_called">>, <<"local">>], Macros),
+
+    Extension = get_extension(MediaJObj),
+    FileName = iolist_to_binary([CallerID, $_, kz_time:pretty_print_datetime(LocalDateTime), $., Extension]),
+
+    kz_http_util:urlencode(
+      binary:replace(kz_term:to_lower_binary(FileName), <<$\s>>, <<$_>>)
+     ).
+
+-spec get_extension(kz_json:object()) -> kz_term:ne_binary().
+get_extension(MediaJObj) ->
+    kz_mime:to_extension(kz_doc:attachment_content_type(MediaJObj)).
+
+-spec build_template_data(kz_json:object()) -> kz_term:proplist().
+build_template_data(DataJObj) ->
+    Timezone = kzd_voicemail_box:timezone(kz_json:get_value(<<"vmbox_doc">>, DataJObj)),
+    [{<<"voicemail">>, build_voicemail_data(DataJObj)}
+    ,{<<"account">>, teletype_util:account_params(DataJObj)}
+    ,{<<"user">>, teletype_util:user_params(kz_json:get_value(<<"user">>, DataJObj))}
+    ,{<<"owner">>, teletype_util:user_params(kz_json:get_value(<<"user">>, DataJObj))}
+    ,{<<"reason">>, render_vm_delete_reason(DataJObj)}
+     | teletype_util:build_call_data(DataJObj, Timezone)
+    ].
+
+-spec build_voicemail_data(kz_json:object()) -> kz_term:proplist().
+build_voicemail_data(DataJObj) ->
+    props:filter_undefined(
+      [{<<"vmbox_id">>, kz_json:get_value(<<"voicemail_box">>, DataJObj)}
+      ,{<<"box">>, kz_json:get_value(<<"voicemail_box">>, DataJObj)} %% backward compatibility
+      ,{<<"vmbox_name">>, kz_json:get_value([<<"vmbox_doc">>, <<"name">>], DataJObj)}
+      ,{<<"vmbox_number">>, kz_json:get_value([<<"vmbox_doc">>, <<"mailbox">>], DataJObj)}
+      ,{<<"msg_id">>, kz_json:get_value(<<"voicemail_id">>, DataJObj)}
+      ,{<<"name">>, kz_json:get_value(<<"voicemail_id">>, DataJObj)} %% backward compatibility
+      ,{<<"transcription">>, get_transcription(DataJObj)}
+      ,{<<"length">>, pretty_print_length(DataJObj)}
+      ]).
+
+-spec render_vm_delete_reason(kz_json:object()) -> kz_term:api_ne_binary().
+render_vm_delete_reason(DataJObj) ->
+    case kz_json:get_atom_value(<<"reason">>, DataJObj) of
+        'dtmf' -> <<"user pressed DTMF key">>;
```

Comment:
I think the reason strings here can be  more user friendly.  How about some thing like this:

* `dtmf`: Delete by calling voicemail box
* `crossbar_action`: Delete from UI

Reply:
@icehess I'm ok with the `dtmf` string as "Deleted by calling voicemail box", however I would suggest for `crossbar_action` to be "Deleted via API" (as it could happen via a system <--> system action, and not just from MonsterUI for instance).

Reply:
@icehess would that work?

Reply:
Yes, "Deleted via API" is better.